mm2src/mm2_bitcoin/spv_validation/src/helpers_validation.rs

```diff
@@ -312,7 +321,63 @@ pub struct BlockHeaderVerificationParams {
     pub difficulty_check: bool,
     pub constant_difficulty: bool,
     pub difficulty_algorithm: Option<DifficultyAlgorithm>,
-    pub genesis_block_header: String,
+    // Where to start fetching block height from. 0 will be the default value if starting_block_header_height is None
+    pub starting_block_header_height: Option<u64>,
+    // Use (starting_block_header_height +1) header hex if starting_block_header_height is set else use genesis block
+    // header hex.
+    pub starting_block_header_hex: String,
+    // retarget_block_header_height is needed if starting_block_header_height is set.
+    pub retarget_block_header_height: Option<u64>,
+    // retarget_block_header_hex is needed if starting_block_header_height is set. (2016)
+    pub retarget_block_header_hex: Option<String>,
+}
+
+#[derive(Debug, Default, Clone, Deserialize)]
+pub struct SPVConf {
+    // Determine if spv proof should be enable. Default to false.
+    pub enable_spv_proof: bool,
+    // Limit of block headers allowed to stored in db. No limit will be set if max_stored_block_headers is None.
+    max_stored_block_headers: Option<u64>,
+    // Where to start fetching block height from. 0 will be the default value if starting_block_header_height is None
+    starting_block_header_height: Option<u64>,
```

Comment:
This field is repeated in `BlockHeaderVerificationParams`

Reply:
yes that's because not every coin will need verification params and I also don't want verification params to rely too much SPVConf itself... in case of this check 
```rust
if let Some(params) = &spv_conf.verification_params {
            if let Err(e) = validate_headers(ticker, from_block_height, block_headers, storage, params).await {
                error!("Error {e:?} on validating the latest headers for {ticker}!");
                // Todo: remove this electrum server and use another in this case since the headers from this server are invalid
                sync_status_loop_handle.notify_on_permanent_error(e.to_string());
                break;
            };
        }
```
`spv_conf.verification_params` needs to know the starting block as well.

Reply:
As a rule, if we have a duplicate field in a nested struct, one of those fields should be removed or the design of the whole struct needs changing.
For `starting_block_header_height`, it is important for both first RPC call and headers verification, so it should be on the outside level only and other code that relies on it should be refactored. It is also related to `starting_retarget_block_header_hex`, so how about having a struct for both in the outside level of the struct, it can be considered a validated header and inserted in the DB straight away, then both won't be required in the `validate_headers` function at all :)

Reply:
good idea.